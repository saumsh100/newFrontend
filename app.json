  {
  "name": "carecru",
  "scripts": {
  },
  "env": {
    "NODE_ENV": {
      "required": true
    },
    "RETHINKDB_AUTHKEY": {
      "required": true
    },
    "RETHINKDB_DB": {
      "required": true
    },
    "RETHINKDB_HOST": {
      "required": true
    },
    "RETHINKDB_PORT": {
      "required": true
    },
    "VENDASTA_API_KEY": {
      "required": true
    },
    "VENDASTA_API_USER": {
      "required": true
    },
    "TWILIO_ACCOUNT_SID": {
      "required": true
    },
    "TWILIO_AUTH_TOKEN": {
      "required": true
    },
    "TWILIO_NUMBER": {
      "required": true
    }
  },
  "formation": {
    "web": {
      "quantity": 1
    }
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
