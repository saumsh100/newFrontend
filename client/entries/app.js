
import '../apps/default';
import '../apps/Dashboard';
