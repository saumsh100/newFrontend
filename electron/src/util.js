
module.exports.getChannelFromVersion = (version) => {
  const splitVersion = version.split('-');
  return splitVersion[1] || null;
};
