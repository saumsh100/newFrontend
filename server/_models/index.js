
const Sequelize = require('sequelize');
const { postgres } = require('../config/globals');

const sequelizeConfig = {
  logging: false,
  dialect: 'postgres',
  host: postgres.host,
  port: postgres.port,
  define: {
    underscored: false,
    paranoid: true,
  },
};

// Don't use NODE_ENV===production
// It doesn't let us debug production locally easily
if (postgres.ssl) {
  sequelizeConfig.dialectOptions = {
    ssl: true,
  };
}

// If true, sequelize will dump all PostgreSQL queries into terminal
if (postgres.logging) {
  sequelizeConfig.logging = console.log // eslint-disable-line
}

// initialize sequelize
const sequelize = new Sequelize(
  postgres.database,
  postgres.username,
  postgres.password,
  sequelizeConfig,
  );

// Test the connection for logs
sequelize
  .authenticate()
  .then(() => {
    console.log(`Sequelize connected to database=${postgres.database} at host:port=${postgres.host}:${postgres.port}`);
  })
  .catch(err => {
    console.error('postgres config', postgres);
    console.error('Unable to connect to the database:', err);
  });

// Import and store all models.
const models = [];
// models.push((require('./Address').default(sequelize, Sequelize)));
models.push((require('./Account').default(sequelize, Sequelize)));
models.push((require('./Appointment').default(sequelize, Sequelize)));
models.push((require('./AuthSession').default(sequelize, Sequelize)));
models.push((require('./Call').default(sequelize, Sequelize)));
models.push((require('./Chair').default(sequelize, Sequelize)));
models.push((require('./Chat').default(sequelize, Sequelize)));
models.push((require('./Enterprise').default(sequelize, Sequelize)));
models.push((require('./Family').default(sequelize, Sequelize)));
models.push((require('./Invite').default(sequelize, Sequelize)));
models.push((require('./Patient').default(sequelize, Sequelize)));
models.push((require('./PatientUser').default(sequelize, Sequelize)));
models.push((require('./Permission').default(sequelize, Sequelize)));
models.push((require('./PinCode').default(sequelize, Sequelize)));
models.push((require('./Practitioner').default(sequelize, Sequelize)));
models.push((require('./Practitioner_Service').default(sequelize, Sequelize)));
models.push((require('./PractitionerRecurringTimeOff').default(sequelize, Sequelize)));
models.push((require('./Recall').default(sequelize, Sequelize)));
models.push((require('./Reminder').default(sequelize, Sequelize)));
models.push((require('./Request').default(sequelize, Sequelize)));
models.push((require('./Segment').default(sequelize, Sequelize)));
models.push((require('./SentRecall').default(sequelize, Sequelize)));
models.push((require('./SentReminder').default(sequelize, Sequelize)));
models.push((require('./Service').default(sequelize, Sequelize)));
models.push((require('./SyncClientError').default(sequelize, Sequelize)));
models.push((require('./SyncClientVersion').default(sequelize, Sequelize)));
models.push((require('./TextMessage').default(sequelize, Sequelize)));
models.push((require('./Token').default(sequelize, Sequelize)));
models.push((require('./User').default(sequelize, Sequelize)));
models.push((require('./WaitSpot').default(sequelize, Sequelize)));
models.push((require('./WeeklySchedule').default(sequelize, Sequelize)));

const db = {};
models.forEach((model) => {
  db[model.name] = model;
});

Object.keys(db).forEach((modelName) => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
