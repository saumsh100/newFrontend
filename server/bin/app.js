
require('../config/initializeCodeTranspiler');

const globals = require('../config/globals');
const applyWebpack = require('../config/webpack/applyWebpack');
const handleErrors = require('../middleware/handleErrors');
const chalk = require('chalk');
const app = require('../config/express');

// require('../config/kue');

// Uses the NODE_ENV to determine logging full stack or not
app.set('showStackError', true);

// Load in webpack configurations
applyWebpack(app);


// Set the Handlebars templating engine
app.set('views', `${globals.root}/views`);
app.engine('hbs', require('../config/handlebars').engine);
app.set('view engine', 'hbs');

// Extra logging for communication with server
app.use((req, res, next) => {
  console.log(chalk.blue(req.method, req.originalUrl));
  next();
});


// Bind the routes

app.use(require('../routes'));

// Catch errors, log and respond to client
app.use(handleErrors);

// assume 404 since no middleware responded
/*app.use((req, res) => {
  res.status(404).render('404', {
    url: req.originalUrl,
    error: 'Not found',
  });
});*/

module.exports = app;
