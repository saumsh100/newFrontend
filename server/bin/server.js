
const http = require('http');
const globals = require('../config/globals');
const app = require('./app');
const Availability = require('../models/Availability');
const Patient = require('../models/Patient');

const server = http.createServer(app);

const io = require('socket.io')(server);

// TODO: implement Thinky!!!
io.on('connection', (socket) => {
  console.log('a new socket connected');
  socket.on('fetchAvailabilities', () => {
    Availability.run()
      .then((availabilities) => {
        socket.emit('receiveAvailabilities', availabilities);
      });
  });
  
  socket.on('fetchPatients', () => {
    Patient.run()
      .then((patients) => {
        socket.emit('receivePatients', patients);
      });
  });
  
  socket.on('addAvailability', (data) => {
    new Availability(data).save().then((availability) => {
      io.sockets.emit('availabilityAdded', availability);
    });
  });
  
  socket.on('removeAvailability', (data) => {
    Availability.get(data.id).then((availability) => {
      availability.delete()
        .then((result) => {
          io.sockets.emit('availabilityRemoved', result);
        });
    });
  });
  
  socket.on('syncClientTrigger', (data) => {
    console.log('sync client triggers!', 'data:', data);
  });
});

server.listen(globals.port, () => {
  console.log(`CareCru HTTP Server is running on port ${globals.port}`);
});
