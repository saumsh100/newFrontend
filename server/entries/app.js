
import '../../client/apps/default';
import '../../client/apps/Dashboard';
