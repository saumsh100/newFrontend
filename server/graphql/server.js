
import { Router } from 'express';
import graphQLHTTP from 'express-graphql';
import { sequelizeAuthMiddleware } from '../middleware/auth';
import schema from './data/schema';

const GraphQLRouter = Router();

const PORT = 4000;

const GraphQLServer = graphQLHTTP({
  schema,
  graphiql: true,
  pretty: true,
  subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`,
});

// Reusing auth middleware as the server is exposed the same way an API would.
GraphQLRouter.all('*', sequelizeAuthMiddleware, GraphQLServer);

export default GraphQLRouter;

