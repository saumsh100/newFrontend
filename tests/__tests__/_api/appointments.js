
import request from 'supertest';
import app from '../../../server/bin/app';
import generateToken from '../../_util/generateToken';
import { Appointment, Account, WeeklySchedule } from '../../../server/_models';
import wipeModel, { wipeAllModels } from '../../_util/wipeModel';
import { accountId, enterpriseId, seedTestUsers, wipeTestUsers } from '../../_util/seedTestUsers';
import { wipeTestPatients } from '../../_util/seedTestPatients';
import { wipeTestPractitioners } from '../../_util/seedTestPractitioners';
import { patientId } from '../../_util/seedTestPatients';
import { appointment, appointmentId, seedTestAppointments } from '../../_util/seedTestAppointments';
import { weeklyScheduleId, seedTestWeeklySchedules } from '../../_util/seedTestWeeklySchedules';
import { practitionerId } from '../../_util/seedTestPractitioners';
import { omitPropertiesFromBody } from '../../util/selectors';

const batchAppointmentId = '04148c65-292d-4d06-8801-d8061cec7b12';
const batchAppointmentId2 = '6ee224d6-8e66-4cb8-9d41-8964a3d9b909';
const batchAppointmentId3 = 'f3905556-45f7-406d-ba5a-dabe02288665';
const batchAppointmentId4 = 'bd7a83d3-c1f2-40c5-8cb9-7e3b4362194a';
const invalidBatchAppointmentId = '61073a69-085c-4646-9dc0-2ad85a0b94fb';

const rootUrl = '/_api/appointments';

const accountWithSchedule = {
  id: accountId,
  enterpriseId,
  name: 'Test Account',
  weeklyScheduleId,
  timezone: 'America/Vancouver',
  createdAt: '2017-07-19T00:14:30.932Z',
};

const batchAppointment = {
  id: batchAppointmentId,
  startDate: '2017-07-21T00:14:30.932Z',
  endDate: '2017-07-21T00:16:30.932Z',
  patientId,
  practitionerId,
  isSyncedWithPms: false,
  isReminderSent: true,
  isDeleted: false,
  createdAt: '2017-07-19T00:14:30.932Z',
};

const batchAppointment2 = {
  id: batchAppointmentId2,
  startDate: '2017-07-21T00:14:30.932Z',
  endDate: '2017-07-21T00:16:30.932Z',
  patientId,
  practitionerId,
  isSyncedWithPms: false,
  isReminderSent: true,
  isDeleted: false,
  createdAt: '2017-07-19T00:14:30.932Z',
};

const batchAppointment3 = {
  id: batchAppointmentId3,
  startDate: '2017-07-21T00:14:30.932Z',
  endDate: '2017-07-21T00:16:30.932Z',
  patientId,
  practitionerId,
  isSyncedWithPms: false,
  isReminderSent: true,
  isDeleted: false,
  createdAt: '2017-07-19T00:14:30.932Z',
};

const batchAppointment4 = {
  id: batchAppointmentId4,
  startDate: '2017-07-21T00:14:30.932Z',
  endDate: '2017-07-21T00:16:30.932Z',
  patientId,
  practitionerId,
  isSyncedWithPms: false,
  isReminderSent: true,
  isDeleted: false,
  createdAt: '2017-07-19T00:14:30.932Z',
};

const invalidBatchAppointment = {
  id: invalidBatchAppointmentId,
  patientId,
  practitionerId,
  isSyncedWithPms: false,
  isReminderSent: true,
  isDeleted: false,
  createdAt: '2017-07-19T00:14:30.932Z',
};

describe('/api/appointments', () => {
  // Seed with some standard user data
  let token = null;
  beforeEach(async () => {
    await seedTestUsers();
    await seedTestAppointments();
    await Account.update({ weeklyScheduleId }, { where: { id: accountId } }).catch(err => console.log(err));
    token = await generateToken({ username: 'manager@test.com', password: '!@CityOfBudaTest#$' });
  });

  afterAll(async () => {
    await wipeModel(Account);
    await wipeModel(WeeklySchedule);
    await wipeModel(Appointment);
    await wipeTestPatients();
    await wipeTestPractitioners();
    await wipeTestUsers();
    await wipeAllModels();
  });

  // TODO: This can use some more test cases... (Gavin: Not familiar with what's going on in these endpoints)
  describe('GET /', () => {
    test('/business - [no description]', () => {
      return request(app)
        .get(`${rootUrl}/business?startDate=2016-07-19T00:14:30.932Z&endDate=2018-07-19T00:14:30.932Z`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/statsdate - data for most popular day of the week', () => {
      return request(app)
        .get(`${rootUrl}/statsdate`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/statslastyear - data for past year', () => {
      return request(app)
        .get(`${rootUrl}/statslastyear`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/stats - appointment stats for intelligence overview', () => {
      return request(app)
        .get(`${rootUrl}/stats?startDate=2016-07-19T00:14:30.932Z&endDate=2018-07-19T00:14:30.932Z`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/ - retrieve appointments', () => {
      return request(app)
        .get(`${rootUrl}?startDate=2016-07-19T00:14:30.932Z&endDate=2018-07-19T00:14:30.932Z`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/ - retrieve appointments - pms not synced', () => {
      return request(app)
        .get(`${rootUrl}/connector/notSynced`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/:appointmentId - retrieve an appointment', () => {
      return request(app)
        .get(`${rootUrl}/${appointmentId}`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/ - retrieve appointments - return nothing - pms not synced', async () => {
      await Appointment.update({ isSyncedWithPms: true }, { where: { id: '6b215a42-5c33-4f94-8313-d89893ae2f36' } });
      return request(app)
        .get(`${rootUrl}/connector/notSynced`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('POST /', () => {
    beforeEach(async () => {
      await wipeModel(Appointment);
    });

    test('/ - create an appointment', () => {
      const createAppointment = Object.assign({}, appointment);
      appointment.accountId;
      return request(app)
        .post(rootUrl)
        .set('Authorization', `Bearer ${token}`)
        .send(createAppointment)
        .expect(201)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/ - create a short Cancel - both cancelled fields should be true', () => {
      const createAppointment = Object.assign({ isShortCancelled: true }, appointment);
      appointment.accountId;
      return request(app)
        .post(rootUrl)
        .set('Authorization', `Bearer ${token}`)
        .send(createAppointment)
        .expect(201)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/ - create a Cancel - only isCancelled field should be true not isShortCancelled', () => {
      const createAppointment = Object.assign({ isCancelled: true }, appointment);
      appointment.accountId;
      return request(app)
        .post(rootUrl)
        .set('Authorization', `Bearer ${token}`)
        .send(createAppointment)
        .expect(201)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });


    test('/batch - 4 appointments created successfully', () => {
      return request(app)
        .post(`${rootUrl}/batch`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          appointments: [batchAppointment, batchAppointment2, batchAppointment3, batchAppointment4],
        })
        .expect(201)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(Object.keys(body.entities.appointments).length).toBe(4);
        });
    });


    test('/batch - 1 invalid appointment, 3 valid appointments', () => {
      return request(app)
        .post(`${rootUrl}/batch`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          appointments: [invalidBatchAppointment, batchAppointment2, batchAppointment3, batchAppointment4],
        })
        .expect(400)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(Object.keys(body.entities.appointments).length).toBe(3);
        });
    });
  });


  describe('PUT /', () => {

    test('/batch - update 1 appointment', () => {
      const updateAppointment = appointment;
      appointment.isReminderSent = false;
      return request(app)
        .put(`${rootUrl}/batch`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          appointments:
            [updateAppointment],
        })
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(Object.keys(body.entities.appointments).length).toBe(1);
          expect(body).toMatchSnapshot();
        });
    });

    test('/:appointmentId - update appointment', () => {
      return request(app)
        .put(`${rootUrl}/${appointmentId}`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          isSyncedWithPms: true,
        })
        .expect(201)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('DELETE /', () => {
    test('/:appointmentId - delete appointment', () => {
      return request(app)
        .delete(`${rootUrl}/${appointmentId}`)
        .set('Authorization', `Bearer ${token}`)
        .expect(204)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/:appointmentId - delete appointment then undelete it', () => {
      return request(app)
        .delete(`${rootUrl}/${appointmentId}`)
        .set('Authorization', `Bearer ${token}`)
        .expect(204)
        .then(({ body }) => {
          const appointmentCreate = {
            startDate: '2017-01-21T00:14:30.932Z',
            endDate: '2017-01-21T00:14:30.932Z',
            patientId,
            practitionerId,
            pmsId: '12',
            isSyncedWithPms: false,
            isReminderSent: true,
            isDeleted: false,
            createdAt: '2017-07-19T00:14:30.932Z',
          };
          return request(app)
                  .post(rootUrl)
                  .set('Authorization', `Bearer ${token}`)
                  .send(appointmentCreate)
                  .expect(201)
                  .then(({ body }) => {
                    body = omitPropertiesFromBody(body);
                    expect(body).toMatchSnapshot();
                  });
        });
    });
  });
});
