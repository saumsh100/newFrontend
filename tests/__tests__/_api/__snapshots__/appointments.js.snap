// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/appointments DELETE / /:appointmentId - delete appointment 1`] = `
Object {
  "entities": Object {},
}
`;

exports[`/api/appointments DELETE / /:appointmentId - delete appointment then undelete it 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": false,
        "isShortCancelled": false,
        "isSyncedWithPms": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": "6b215a42-5c33-4f94-8313-d89893ae2f36",
}
`;

exports[`/api/appointments GET / / - retrieve appointments - pms not synced 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": true,
        "isShortCancelled": false,
        "isSyncedWithPms": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": Array [
    "6b215a42-5c33-4f94-8313-d89893ae2f36",
  ],
}
`;

exports[`/api/appointments GET / / - retrieve appointments - return nothing - pms not synced 1`] = `
Object {
  "entities": Object {},
  "result": Array [],
}
`;

exports[`/api/appointments GET / / - retrieve appointments 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": true,
        "isShortCancelled": false,
        "isSyncedWithPMS": false,
        "isSyncedWithPms": false,
        "mark": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": Array [
    "6b215a42-5c33-4f94-8313-d89893ae2f36",
  ],
}
`;

exports[`/api/appointments GET / /:appointmentId - retrieve an appointment 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": true,
        "isShortCancelled": false,
        "isSyncedWithPMS": false,
        "isSyncedWithPms": false,
        "mark": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": "6b215a42-5c33-4f94-8313-d89893ae2f36",
}
`;

exports[`/api/appointments GET / /business - [no description] 1`] = `
Object {
  "brokenAppts": 0,
  "entities": Object {},
  "hygieneAppts": 1,
  "productionEarnings": Array [],
}
`;

exports[`/api/appointments GET / /stats - appointment appointmentsBooked for intelligence overview 1`] = `
Object {
  "appointmentsBooked": "1",
  "confirmedAppointments": 0,
  "entities": Object {},
}
`;

exports[`/api/appointments GET / /stats - appointment mostAppointments for intelligence overview 1`] = `
Array [
  Object {
    "birthDate": "1980-01-01T00:00:00.932Z",
    "firstName": "Ronald",
    "id": "10518e11-b9d2-4d74-9887-29eaae7b5938",
    "lastName": "Mcdonald",
    "numAppointments": "1",
  },
]
`;

exports[`/api/appointments GET / /stats - appointment practitionerWorked for intelligence overview 1`] = `
Array [
  Object {
    "avatarUrl": null,
    "booked": 0,
    "firstName": "Colonel",
    "id": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
    "lastName": "Sanders",
    "newPatientsTotal": 0,
    "notFilled": 6570,
    "type": "Hygienist",
  },
]
`;

exports[`/api/appointments GET / /stats - appointment stats for intelligence overview 1`] = `
Object {
  "activePatients": 1,
  "entities": Object {},
  "newPatients": 0,
}
`;

exports[`/api/appointments GET / /statsdate - data for most popular day of the week 1`] = `
Object {
  "days": Array [
    0,
    0,
    0,
    0,
    1,
    0,
    0,
  ],
  "entities": Object {},
}
`;

exports[`/api/appointments GET / /statslastyear - data for past year 1`] = `
Object {
  "age": Array [
    0,
    0,
    0,
    1,
    0,
    0,
  ],
  "data": Array [
    0,
    0,
    0,
    0,
    1,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "entities": Object {},
  "months": Array [
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
    "January",
    "February",
  ],
}
`;

exports[`/api/appointments GET /insights /insights - appointment insights 1`] = `
Object {
  "insights": Array [
    Object {
      "type": "missingMobilePhone",
      "value": null,
    },
    Object {
      "type": "missingEmail",
      "value": null,
    },
  ],
}
`;

exports[`/api/appointments POST / / - create a Cancel - only isCancelled field should be true not isShortCancelled 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": true,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": true,
        "isShortCancelled": false,
        "isSyncedWithPMS": false,
        "isSyncedWithPms": false,
        "mark": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": "6b215a42-5c33-4f94-8313-d89893ae2f36",
}
`;

exports[`/api/appointments POST / / - create a short Cancel - both cancelled fields should be true 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": true,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": true,
        "isShortCancelled": true,
        "isSyncedWithPMS": false,
        "isSyncedWithPms": false,
        "mark": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": "6b215a42-5c33-4f94-8313-d89893ae2f36",
}
`;

exports[`/api/appointments POST / / - create an appointment 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": true,
        "isShortCancelled": false,
        "isSyncedWithPMS": false,
        "isSyncedWithPms": false,
        "mark": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": "6b215a42-5c33-4f94-8313-d89893ae2f36",
}
`;

exports[`/api/appointments PUT / /:appointmentId - update appointment 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": false,
        "isShortCancelled": false,
        "isSyncedWithPMS": true,
        "isSyncedWithPms": true,
        "mark": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": "6b215a42-5c33-4f94-8313-d89893ae2f36",
}
`;

exports[`/api/appointments PUT / /batch - update 1 appointment 1`] = `
Object {
  "entities": Object {
    "appointments": Object {
      "6b215a42-5c33-4f94-8313-d89893ae2f36": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "chairId": null,
        "customBufferTime": null,
        "endDate": "2017-07-21T00:16:30.932Z",
        "estimatedRevenue": null,
        "id": "6b215a42-5c33-4f94-8313-d89893ae2f36",
        "isBookable": false,
        "isCancelled": false,
        "isDeleted": false,
        "isPatientConfirmed": false,
        "isPending": false,
        "isPreConfirmed": null,
        "isReminderSent": false,
        "isShortCancelled": false,
        "isSyncedWithPMS": false,
        "isSyncedWithPms": false,
        "mark": false,
        "note": null,
        "patientId": "10518e11-b9d2-4d74-9887-29eaae7b5938",
        "pmsId": "12",
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "reason": null,
        "serviceId": null,
        "startDate": "2017-07-21T00:14:30.932Z",
      },
    },
  },
  "result": Array [
    "6b215a42-5c33-4f94-8313-d89893ae2f36",
  ],
}
`;
